components/org.wso2.carbon.identity.local.auth.smsotp.authenticator/src/main/java/org/wso2/carbon/identity/local/auth/smsotp/authenticator/SMSOTPAuthenticator.java

```diff
@@ -325,7 +326,19 @@ protected void sendOtp(AuthenticatedUser authenticatedUser, OTP otp, boolean isI
                 String.valueOf(getOtpValidityPeriodInMillis(authenticationContext.getTenantDomain()) / 60000));
         metaProperties.put(SMSOTPConstants.TEMPLATE_TYPE, SMSOTPConstants.EVENT_NAME);
         setAuthenticatorMessage(authenticationContext, mobileNumber);
-        triggerEvent(SMSOTPConstants.EVENT_TRIGGER_NAME, authenticatedUser, metaProperties);
+        /* SaaS apps are created at the super tenant level and they can be accessed by users of other organizations.
+        If users of other organizations try to login to a saas app, the sms notification should be triggered from the
+        sms provider configured for that organization. Hence, we need to start a new tenanted flow here. */
+        if (authenticationContext.getSequenceConfig().getApplicationConfig().isSaaSApp()) {
```

Comment:
Do we really need to start a tenant flow here? Does it require for the execution of the flow that is running after this? If we just need the tenant of the user, can't we pass it as a parameter and use?

Reply:
I have refer the same as done for email otp authenticator in https://github.com/wso2-extensions/identity-local-auth-emailotp/blob/ab48d011c47a01e639989466bba6ba9811a5fda5/components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java#L714

Reply:
My though was whether this is a overkill. (Since this an expensive operation). However, I think the proper mechanism should be starting a tenant flow as that is what expected in latter stages as well. +1 for this.